src/main/kotlin/com/ampnet/web3provider/service/impl/AccountInformationServiceImpl.kt

```diff
@@ -23,16 +25,18 @@ class AccountInformationServiceImpl(
         val blockParameter = request.params.getOrNull(1) ?: DefaultBlockParameterName.LATEST.value
         logger.info { "Received request to get ${request.method} for address: $address and block: $blockParameter" }
         val hashKey = address?.let { it.toString() + blockParameter }
-        redisRepository.getCache(RedisEntity.BALANCE.methodName, hashKey)?.let { return ProviderResponse(request, it) }
-        return defaultProviderService.getResponseAndUpdateCache(request, RedisEntity.BALANCE, hashKey)
+        val balance = RedisEntity.Balance(applicationProperties)
+        redisRepository.getCache(balance.methodName, hashKey)?.let { return ProviderResponse(request, it) }
```

Comment:
I don't like this implementation because you need to provide `applicationProperties` for a call to get from the cache. TTL value should not be connected to get calls, only store calls. 

I would recommend to provided cache TTL values only in update calls. There are 2 solutions: provide application properties to `DefaultProviderService` or `RedisRepository`.

Reply:
Yeah my implementation is overly complicated and created unnecessary dependancy of all the services to application properties.